mods/creative/inventory.lua

```diff
@@ -59,22 +59,34 @@ function creative.init_creative_inventory(player)
 	return player_inventory[player_name]
 end
 
+local function match(s, filter)
+	if filter == "" then return 0 end
+	if s:lower():find(filter, 1, true) then
+		return #s - #filter
+	end
+	return nil
+end
+
 function creative.update_creative_inventory(player_name, tab_content)
-	local creative_list = {}
 	local inv = player_inventory[player_name] or
 			creative.init_creative_inventory(minetest.get_player_by_name(player_name))
 	local player_inv = minetest.get_inventory({type = "detached", name = "creative_" .. player_name})
 
 	local items = inventory_cache[tab_content] or init_creative_cache(tab_content)
 
+	local creative_list = {}
+	local order = {}
 	for name, def in pairs(items) do
-		if def.name:find(inv.filter, 1, true) or
-				def.description:lower():find(inv.filter, 1, true) then
+		local m = match(def.description, inv.filter) or match(def.name, inv.filter)
+		if m then
 			creative_list[#creative_list+1] = name
+			-- Sort by description length first so closer matches appear earlier
+			order[name] = string.format("%02d", m) .. name
```

Comment:
There is probably a good reason for this but, why not just :
`order[name] = m`
?
I would have thought
`return order[a] < order[b]`
would work with the table entries just being numbers.

Reply:
I want to sort by item name too, if match() returns 0.
Might work regardless since the list is already sorted beforehand, but I don't know if Lua guarantees that.

Reply:
I still want to sort by item name when match() returns 0.
Might work regardless since the table is already sorted beforehand, but I don't know if Lua guarantees that.

Reply:
Ah i see, fine.